l1/ecl_l1_pos_controller.cpp

```diff
@@ -45,11 +45,27 @@
 using matrix::Vector2f;
 using matrix::wrap_pi;
 
-float ECL_L1_Pos_Controller::nav_roll()
+
+void ECL_L1_Pos_Controller::update_roll_setpoint()
 {
-	float ret = atanf(_lateral_accel * 1.0f / CONSTANTS_ONE_G);
-	ret = math::constrain(ret, -_roll_lim_rad, _roll_lim_rad);
-	return ret;
+	float roll_new = atanf(_lateral_accel * 1.0f / CONSTANTS_ONE_G);
+	roll_new = math::constrain(roll_new, -_roll_lim_rad, _roll_lim_rad);
+
+	// no slew rate limiting active
+	if (_dt <= 0.0f || _roll_slew_rate <= 0.0f) {
```

Comment:
What about storing only the last timestamp within L1 and have it calculate its own dt as needed?

Reply:
@dagar Yeah I was wondering about that too. I checked TECS and that's apparently the way you do it.

Reply:
It's not necessarily "the right way", but avoiding the need for multiple calls into something in the same iteration certainly makes the API harder to use incorrectly.

Reply:
That or grab its own timestamp internally when called. L1 should either track its own state to do this independently or stick to the simple return. It's nitpicky at this point, but I think we should try to avoid letting the API become awkward (eg tecs) if anyone is ever possibly going to use this outside of PX4.